src/views/viewer/DiscoverView/DiscoverView.tsx

```diff
@@ -25,22 +26,28 @@ export const DiscoverView: React.FC = () => {
   }))
 
   const categoriesFeaturedVideos = useCategoriesFeaturedVideos()
-  const featuredVideoCategoryCardsData = categories
-    ?.map((category) => {
-      const video =
-        categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl) ??
-        categoriesFeaturedVideos?.[category.id]?.[0] ??
-        undefined
-      return video === undefined
-        ? undefined
-        : {
-            videoTitle: video?.title ?? '',
-            videoUrl: video?.videoCutUrl ?? '',
-            ...mappedVideoCategories?.find((cat) => cat.id === category.id),
-          }
-    })
-    .filter((cat) => !!cat) ?? [null, null, null]
+  const featuredVideoCategoryCardsData = React.useMemo(() => {
+    const _featuredVideoCategoryCardsData = sampleSize(
+      categories
+        ?.map((category) => {
+          const video =
+            categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl) ??
+            categoriesFeaturedVideos?.[category.id]?.[0] ??
+            undefined
+          return video === undefined
+            ? undefined
+            : {
+                videoTitle: video?.title ?? '',
+                videoUrl: video?.videoCutUrl ?? '',
+                ...mappedVideoCategories?.find((cat) => cat.id === category.id),
+              }
```

Comment:
```suggestion
          if (video) {
            return {
              videoTitle: video?.title ?? '',
              videoUrl: video?.videoCutUrl ?? '',
              ...mappedVideoCategories?.find((cat) => cat.id === category.id),
            }
          }
```

Reply:
I rather keep it explicit that we are possibly returning undefined here

Reply:
Hmm, I don't agree with this. I mean, it's quite clear for me, that if we don't return from `Array.map` we will just return `undefined`. Your current code gives me the impression that something else going on here, and I need to decode this every time I'm looking at these lines 😅 . 
Of course, this is only my opinion. Not sure if everyone on the team agrees with me. Let's leave that unresolved for now, maybe?

Reply:
I think we actually shouldn't rely on `undefined` that much, `null` seems to be a better fit here as it explicitly tells you the value is missing. How about this, should be both readable and explicit:
```tsx
const video =
  categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl) ??
  categoriesFeaturedVideos?.[category.id]?.[0]

if (!video) return null

return {
  // ...
}
```